src/util/auth/google_authentication_repository.ts

```diff
@@ -23,19 +20,20 @@ const save = (accessToken: string, refreshToken: string): void => {
 const authenticate = async (
   authorizationCode: string
 ): Promise<AuthenticationResponse> => {
-  const authenticationURL = new URL("https://oauth2.googleapis.com/token");
-  authenticationURL.search = Object.keys({
-    client_id: import.meta.env.VITE_GOOGLE_CLIENT_ID,
-    client_secret: import.meta.env.VITE_GOOGLE_CLIENT_SECRET,
-    code: authorizationCode,
-    grant_type: "authorization_code",
-    redirect_uri: import.meta.env.VITE_HOSTURL,
-  })
-    .map((key, val) => `${key}=${val}`)
```

Comment:
@leekeh `val` here is the index, instead of its value. thats why you were getting errors trying to log in 😄 

If you know how to fix it, feel free to do it/recommend it, for now am just using a map instead